{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cotools as co\n",
    "from pprint import pprint\n",
    "import os\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rootDir = '.'\n",
    "# for dirName, subdirList, fileList in os.walk(rootDir):\n",
    "#     print('Found directory: %s' % dirName)\n",
    "#     for fname in fileList:\n",
    "#         print('\\t%s' % fname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "64\n",
      "376\n"
     ]
    }
   ],
   "source": [
    "comm_use = co.Paperset(\"data/comm_use_subset/comm_use_subset\")\n",
    "print(sys.getsizeof(comm_use))\n",
    "print(sys.getsizeof(comm_use[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9118\n",
      "<class 'dict'>\n"
     ]
    }
   ],
   "source": [
    "print(len(comm_use))\n",
    "print(type(comm_use[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "536\n",
      "<class 'list'>\n",
      "[<class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>, <class 'dict'>]\n"
     ]
    }
   ],
   "source": [
    "print(sys.getsizeof(comm_use[15:65]))\n",
    "print(type(comm_use[15:65]))\n",
    "print([type(x) for x in comm_use[15:65]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Porcine epidemic diarrhea virus (PEDV), which belongs to the '\n",
      " 'Alphacoronavirus genus of the Coronaviridae family, is an etiological agent '\n",
      " 'of porcine epidemic diarrhea (PED) and causes an enteric disease that '\n",
      " 'affects all ages of swine [1, 2] . The clinical presentations and '\n",
      " 'complications of infection are characterized by acute vomiting, dehydration, '\n",
      " 'watery diarrhea, and high mortality in sucking piglets [3] and are '\n",
      " 'indistinguishable from those of infection by either transmissible '\n",
      " 'gastroenteritis virus (TGEV) or porcine enteric alphacoronavirus (PEAV) [4, '\n",
      " '5] . First detected in the UK in 1971, PEDV resulted in mass epidemics '\n",
      " 'within Europe in the 1970s and 1980s [6] . Before 2013, PED was prevalent in '\n",
      " 'Asia and Europe [1] . After spring 2013, however, PED outbreaks reached '\n",
      " 'North America, which was due to variant PEDV strains that researchers '\n",
      " 'revealed might derive from Chinese variants [7, 8] . In spite of widespread '\n",
      " 'immunization with the currently marketed vaccine, PED still persists in '\n",
      " 'swine raising countries and resulted in devastating damage to the pork '\n",
      " 'producers [9] . PEDV is an enveloped single-strand RNA coronavirus with a 28 '\n",
      " 'kb genome, which includes 4 open reading frames encoding spike (S), envelope '\n",
      " '(E), membrane (M), nucleocapsid (N), as well as 3 open reading frames '\n",
      " 'encoding replicase 1a, 1b and ORF3 [10] . As known for other coronaviruses, '\n",
      " 'the three PEDV S glycoproteins form a club-shaped functional S trimer, which '\n",
      " 'is localized on the surface of the virion and mediates essential biological '\n",
      " 'functions, such as membrane fusion and receptor binding. The S protein is '\n",
      " 'also responsible for the induction of nAbs and protective immunity, making '\n",
      " 'it an appropriate candidate for developing an effective vaccine and '\n",
      " 'diagnostic reagents [1, 11, 12] . In addition, variation in the S gene leads '\n",
      " 'to antigenic diversity, and thus the S protein is useful in evaluating '\n",
      " 'genetic diversity [13] . Little has been known about the components of the '\n",
      " 'immune system that are effective in the protection of a pig against PEDV '\n",
      " 'infection. The quantity of nAbs generated by vaccination correlates with the '\n",
      " 'degree of protection against many diseases [14] . Considering the '\n",
      " 'significance of nAbs in providing protection, understanding the mechanism of '\n",
      " 'neutralization is necessary for development of a vaccine that elicits strong '\n",
      " 'nAbs. The fragment antigen-binding (Fab) domain binds to specific pathogen '\n",
      " 'targets, which prevents microbial interactions with host cell receptors and '\n",
      " 'thus blocks infection [15, 16] . The protection of nAbs results from '\n",
      " 'blocking interaction of free virus particles with target cell receptors. '\n",
      " 'Additionally, for other nAbs, infection can be blocked through inhibiting '\n",
      " 'critical intracellular processes, for example rotavirus transcription [17] , '\n",
      " 'nuclear translocation of human papilloma virus DNA [18] , adenoviral '\n",
      " 'uncoating [19] , or measles virus assembly [20] . Several studies '\n",
      " 'demonstrated that spike mAb can neutralize PEDV [21] [22] [23] [24] [25] '\n",
      " '[26] . These studies mainly focused on locating the neutralizing domains of '\n",
      " 'PEDV S protein, however, the mechanisms by which spike nAb neutralize the '\n",
      " 'virus have not been defined completely. To fill this knowledge gap, we '\n",
      " 'generated four mAbs that exhibited potent neutralizing activity against PEDV '\n",
      " 'in vitro. Notably, 2B11 and 2G8 were found to block PEDV entry into Vero '\n",
      " 'cell. The Vero E6 cell line was cultured and maintained at 37°C in DMEM '\n",
      " 'containing 10% FBS and antibiotics (100 U/mL of penicillin and 100 μg/mL of '\n",
      " 'streptomycin) (Solarbio, Beijing, China). Sf9 insect cell line was '\n",
      " 'maintained as suspension in serum-free SF900II medium at 27°C in spinner '\n",
      " 'flasks at a speed of 90 to 100 rpm. PEDV-GDS01 (KM089829.1) and PEDV-GDS03 '\n",
      " '(AB857235.1) were propagated in Vero cells with 10 μg/mL trypsin. PEDV '\n",
      " 'strain used in this study indicated GDS01 strain unless otherwise noted. '\n",
      " 'PEDV was propagated and purified as in previously described [27] . Briefly, '\n",
      " 'Vero cells were washed twice with phosphate buffered saline (PBS) to remove '\n",
      " 'residual DMEM, followed by 1 h incubation with PEDV at 37°C and wash with '\n",
      " 'PBS. Next, the cells were infected by the virus via addition of DMEM '\n",
      " 'containing 10 μg/mL of trypsin. The cells were harvested at 36 h after '\n",
      " 'infection, when all cells showed characteristic cytopathogenic effect. Three '\n",
      " 'cycles of freeze-thaws were done to release the intracellular virus '\n",
      " 'particles, and a 30-min centrifugation at 10000 g was performed to pellet '\n",
      " 'cellular debris. After clarification, the supernatant was enriched 100 times '\n",
      " 'by the ultracentrifugation at 30000 g and then purified by sucrose density '\n",
      " 'gradient centrifugation using sucrose solutions at: 20%(w/w), 40% (w/w) and '\n",
      " '60% (w/ w), respectively. The purified products were analyzed by SDS-PAGE '\n",
      " 'and western blot. Standard procedures were used to generate hybridoma cells '\n",
      " 'that secrete PEDV-specific antibodies [28] with some modifications. Briefly, '\n",
      " 'female BALB/c mice (6 weeks) were immunized with the purified PEDV '\n",
      " \"inactivated by β-propiolactone in complete Freund's adjuvant (Sigma, St. \"\n",
      " 'Louis, MO, USA). The mouse was immunized with the purified PEDV containing '\n",
      " 'of 10 μg spike protein determined through SDS-PAGE and gray scanning. Two '\n",
      " 'booster immunizations were administered at 2-week intervals with PEDV in '\n",
      " \"incomplete Freund's adjuvant (Sigma, St. Louis, MO, USA). Next, the mice \"\n",
      " 'were sacrificed following a 3-day booster inoculation by intraperitoneal '\n",
      " 'injection. PEG1450 [50% (v/v)] (Sigma, St. Louis, MO, USA) was used for '\n",
      " 'fusion of spleen cells from immunized mice with sp2/0 myeloma cells, and '\n",
      " 'hybridoma cells was cultured in 96-well plates at 37°C in HAT (Sigma, St. '\n",
      " 'Louis, MO, USA) screening culture medium. Positive hybridoma clones were '\n",
      " 'picked by indirect immunofluorescence assay (IFA), followed by cloning via '\n",
      " 'limiting dilution for at least three rounds. Polyclonal antibodies against '\n",
      " 'PEDV were taken as positive control and normal mouse serum was taken as a '\n",
      " 'negative control. Mouse Monoclonal Antibody Isotyping Reagents (Sigma, St. '\n",
      " 'Louis, MO, USA) were used for the identification of the subtype of mAbs '\n",
      " 'secreted by the final hybridoma clones. Ascites fluid was collected from '\n",
      " 'primed BALB/c mice with paraffin oil and purified using Protein G Sepharose™ '\n",
      " \"4 Fast Flow (GE Healthcare, Pittsburgh, USA) according to the manufacturer's \"\n",
      " 'instructions. Purified mAb was quantified by BCA kit (Thermo fisher, USA). '\n",
      " 'The supernatant of hybridoma cell cultures was screened for the presence of '\n",
      " 'PEDV-specific mAbs by IFA. For this, primary Vero cells were grown to 100% '\n",
      " 'confluency in 96 well plates and infected with GDS01 for 36 h at a '\n",
      " 'multiplicity of infection (MOI) of 0.1. After fixation with 4% '\n",
      " 'paraformaldehyde, the monolayers were permeabilized with 0.5% triton X-100, '\n",
      " 'followed by a 1 h incubation with the supernatants of the hybridoma cell at '\n",
      " '37°C. Then, unbound antibodies were removed by washing with PBS and specific '\n",
      " 'mAbs were detected with Cy3-conjugated Affinipure Goat Anti-Mouse IgG (H + '\n",
      " 'L) (Proteintech, Rosemont, USA). In order to determine the conformational '\n",
      " 'epitopes bound by the mAbs, the main structural proteins of PEDV, including '\n",
      " 'SP (the S1 and partly S2 gene fragment, 1-954aa), N, M, and ORF3, were '\n",
      " 'expressed by Bac-to-Bac expression system (Invitrogen Carlsbad, CA) '\n",
      " \"following the manufacturer's instructions. The sf9 cells were infected with \"\n",
      " 'recombinant baculovirus (MOI = 5), followed by fixation, permeabilization '\n",
      " 'and incubation with supernatants of the hybridoma cells 3 days later, '\n",
      " 'reactivity of the mAbs with recombinant proteins was measure with an IFA. In '\n",
      " 'order to determine the linear epitopes bound by the mAbs, truncated SP and '\n",
      " 'full length of N genes were also cloned into pET-32a, the details of 7 '\n",
      " 'truncated SP proteins refer to previous study [29] . The recombinant DNA was '\n",
      " 'then used to transform BL21 cells for the following protein expression. '\n",
      " \"Referring to the manufacturer's instructions, Ni-Chelating Sepharose Fast \"\n",
      " 'Flow (GE, USA) was used for the purification of proteins by affinity '\n",
      " 'chromatography. Purified protein was quantified by BCA kit (Thermo fisher, '\n",
      " 'USA). The reaction of mAbs with truncated SP and N protein was evaluated by '\n",
      " 'ELISA. Briefly, 96-well plates (Griener, Germany) were coated with the '\n",
      " 'purified protein (100 ng) at 4°C overnight, and then blocked with 5% milk '\n",
      " 'for 1 h. After washing three times with PBS, 100 μL supernatant was added '\n",
      " 'and the sample was incubated at 37°C for 1 h. Subsequently, the plates were '\n",
      " 'washed with PBS and incubated with HRP-conjugated goat anti-mouse IgG '\n",
      " '(Proteintech, USA) at 37°C for 1 h. The absorbance was measured at 450 nm. '\n",
      " 'All samples were repeated three times and the sample was considered positive '\n",
      " 'when the relation OD sample/OD negative control was higher than 2.1. To '\n",
      " 'determine whether an antibody had neutralization activity, we conducted the '\n",
      " 'virus neutralization test as previously described [30] , with modifications. '\n",
      " 'Briefly, after a 30 min inactivation at 56°C, the test mAbs (diluted to 80 '\n",
      " 'μg/mL) were filtered using a 0.22-μm membrane, followed by two-fold serial '\n",
      " 'dilution. The PEDV GDS01 strain (titer: 100pfu/0.5 mL) was mixed with '\n",
      " 'diluted mAb of an equal volume. The mixture was then added with trypsin (10 '\n",
      " 'μg/mL), followed by 1 h incubation at 37°C. Next, Vero cell monolayers in '\n",
      " '6-well plates were cultured with the mixture (1 mL). After a 1 h adsorption '\n",
      " 'at 37°C, the inocula were discarded. Next, the plates were washed three '\n",
      " 'times with PBS. DMEM with trypsin (10 μg/mL) was added to each well and '\n",
      " 'plates were incubated at 37°C for 48 h. The plaque was colored by the '\n",
      " 'neutral red (0.03%). The Serum-neutralization (SN) titer was determined '\n",
      " 'according to the highest mAb dilution, which led to inhibition of formation '\n",
      " 'of viral plaque completely. Neutralization (%) were calculated using the '\n",
      " 'following formula: 1-sample plaque counts/negative control counts. To '\n",
      " 'determine whether 2G8 and 2B11 could neutralize the infection of GDS03, an '\n",
      " 'IFA neutralization assay was performed. 80 μg of mAb was incubated with an '\n",
      " 'equal volume of 500 TCID 50 /mL PEDV for 1 h at 37°C. Then, the sample-virus '\n",
      " 'mixture was transferred to duplicate wells of a 6-well plate containing '\n",
      " 'confluent Vero E6 cells. The plates were incubated at 37°C for 1 h and then '\n",
      " 'washed gently with PBS to remove unbound viruses, following with 36 h '\n",
      " 'incubation at 37°C in a 5% CO 2 atmosphere. The PEDV-infected cells were '\n",
      " 'fixed with 4% paraformaldehyde and analysed by IFA. 9G11 was used as a '\n",
      " 'detective antibody. Virus infection in the presence or absence of antibody '\n",
      " 'was quantified as previously described, with slight modifications [22] . '\n",
      " 'Diluted antibodies (2B11/100 μg/mL and 2G8/ 200μg/mL) were mixed with PEDV '\n",
      " '(1000pfu/mL) of an equal volume, followed by a 1 h incubation at 37°C. Next, '\n",
      " 'the mixture of antibody-virus was added to triplicate wells of confluent '\n",
      " 'Vero E6 cell monolayers for a 1 h infectious adsorption at 37°C. To analyze '\n",
      " 'PEDV and nAb binding at two different time points, PEDV was incubated with '\n",
      " 'Vero E6 cells for one hour at 4°C, followed by addition of antibodies and a '\n",
      " '1 h incubation at 37°C. Trypsin was added throughout the experiment. The '\n",
      " 'cells were washed twice with PBS, and collected for measuring '\n",
      " 'cell-associated PEDV via viral RNA RT-qPCR. Briefly, the cells from each '\n",
      " 'well were obtained after centrifugation at 10,000 rpm for 10 min. RNA was '\n",
      " 'extracted from cells using a TRIzol reagent (Invitrogen, USA) and cDNA was '\n",
      " 'synthesized with 2 μg of RNA using RT-PCR kit (TaKaRa, China). The specific '\n",
      " \"primers (sense: 5'-GAATTCCCAAGGGCGAAAA T-3′; antisense: \"\n",
      " \"5'-TTTTCGACAAATTCCGCATCT-3′) and probes (5'-FAM-CGTAGCAGGCTTGCTTCGG \"\n",
      " 'ACCCA-BHQ-3′) were designed to amplify and detect the n gene of PEDV. '\n",
      " 'Real-time PCR assays were carried out in 20 μL reaction mixture containing '\n",
      " '10 μL of Thunderbird Probe qPCR Mix, 1 μL of cDNA template, 0.04 μL of 50× '\n",
      " 'Rox reference dye, 0.2 μM of probe, and 0.3 μL of primers. The PCR '\n",
      " 'amplification was performed with an Applied Biosystem 7500 Fast instrument '\n",
      " '(Life Technologies, USA) under the following conditions: 95°C for 20 s for '\n",
      " 'initial denaturation followed by 40 cycles of 95°C for 3 s and 60°C for 30 '\n",
      " 's. Ten-fold serial dilutions of standard plasmid pET-19 T-N, ranging from 10 '\n",
      " '7 to 10 2 copies/μL, were tested in five replicates with real-time RT-PCR to '\n",
      " 'generate the standard curve. Ninety six-well plates (Griener, Frickenhausen, '\n",
      " 'Germany) were firstly coated by PEDV at a density of 10 6 virions/well at '\n",
      " '4°C overnight, followed by a 1 h block with 4% BSA. After washed with PBS '\n",
      " 'for three times and added with 100 μL diluted mAb (1:100), the sample was '\n",
      " 'incubated for 1 h at 37°C. Next, the plates were rinsed with PBS and '\n",
      " 'cultured for 1 h with mAb 2G8 which labelled with horseradish peroxidase by '\n",
      " 'EZ-Link Maleimide Activated Horseradish Peroxidase Kit (Thermo scientific, '\n",
      " 'USA). The reacting results were visualized using tramethylbenzidine (TMB), '\n",
      " 'stopped by HCL. All samples were repeated three times. The absorbance was '\n",
      " 'determined using a microplate reader (Bio-Tek) at 450 nm. PEDV concentration '\n",
      " 'was enriched 100 fold and purified by sucrose density gradient '\n",
      " 'centrifugation. The purified products were analyzed by SDS-PAGE and western '\n",
      " 'blot, and the results showed that the main structural proteins were reserved '\n",
      " 'after ultracentrifugation (see Additional file 1: Figure S1 ). BALB/c mice '\n",
      " 'were immunized with purified PEDV and ten mAbs were prepared via lymphocyte '\n",
      " 'hybridoma technique, which could react well with the PEDV-infected Vero '\n",
      " 'cells by IFA. As shown in Fig. 1 , the syncytium formed due to PEDV '\n",
      " 'infection of cells were not detected using negative mouse serum, but were '\n",
      " 'specifically stained with the mouse polyclonal antibodies and 10 mAbs. All '\n",
      " 'mAbs were determined to be IgG1/kappa isotype, with the exception of 1A5 '\n",
      " '(IgG3/kappa isotype) (data not show). The PEDV S protein is the only '\n",
      " 'identified target of PEDV nAbs. In order to further determine whether PEDV S '\n",
      " 'or N protein is recognized by the mAb, the 7 truncated SP fragment '\n",
      " '(1-954aa), and full length of N were amplified, inserted into pET-32a and '\n",
      " 'transformed into BL21cells. The expressed proteins were purified using a '\n",
      " 'Ni-column and coated to the enzyme plate with 50 μg per well. ELISA results '\n",
      " 'demonstrated that 9G11 had reactivity with N protein and 3F10 reacted with '\n",
      " \"SP protein. All other mAbs didn't react with the recombinant proteins (Fig. \"\n",
      " '2) . SP, N, M and ORF3 genes were also cloned into pFast-Bac 1 vector. '\n",
      " 'Later, the sequencing results verified the expression plasmid containing SP, '\n",
      " 'N, M and ORF3 in the correct direction and reading code frame, respectively. '\n",
      " 'According to IFA analysis, the recombinant proteins of N, M and SP were '\n",
      " 'successfully expressed in sf9 cells using mouse anti-PEDV polyclonal serum '\n",
      " 'as primary antibody (see Additional file 2: Figure S2 ). shows 2B11, 1E3, '\n",
      " '2B5, 2G10, 1A5, and 3F10 have reactivity with SP protein, 9G11 and 1D11 have '\n",
      " 'reactivity with N protein, 3D9 and 2G8 have no reactivity with any protein. '\n",
      " 'The neutralizing activity of selected specific mAbs was assessed by PRN '\n",
      " 'assays. For this, all mAbs were purified from ascites fluid by protein G and '\n",
      " 'diluted to a working concentration of 80 μg/mL (in PBS). Two-fold serial '\n",
      " \"dilution of the nAbs' working stocks(80 μg/mL-2.5 μg/mL) were tested in \"\n",
      " 'triplicate by PRN assays. As shown in Fig. 4 , 2B11, 2G8, and 1E3 completely '\n",
      " 'neutralized the infection of GDS01 in Vero E6 cells at a working '\n",
      " 'concentration of 10 μg/mL, 20 μg/mL and 80 μg/mL, respectively. mAb 3D9 only '\n",
      " 'neutralized the 58% of the input viruses at the maximum working '\n",
      " \"concentration of 80 μg/mL (Fig. 4a) . All other mAbs didn't have the ability \"\n",
      " 'to neutralize GDS01 virus. In order to evaluate the neutralization effect of '\n",
      " 'nAbs with the different subtype of viruses, 2G8 and 2B11 with the strongest '\n",
      " 'neutralizing capacity were chosen to test the inhibition of GDS03 infection '\n",
      " 'by IFA using 9G11 N protein antibody. GDS03 strain belonging to the G2 '\n",
      " \"subtype doesn't form plaque. As shown in Fig. 4b, 2G8 and 2B11 completely \"\n",
      " 'suppressed the GDS03 infection. These results indicated that 2G8 and 2B11 '\n",
      " 'were able to inhibit the infection of both genogroups of PEDV. To explore '\n",
      " 'mAbs 2B11 and 2G8 neutralizes PEDV infection in or out the cells, we tested '\n",
      " 'them at the lowest concentrations. GDS01 was incubated for 1 h with '\n",
      " 'antibodies at 37°C before the mixture was added to Vero cells. At 1 h '\n",
      " 'post-infection at 37°C, the unbound viruses were washed and the '\n",
      " 'cell-associated PEDVs were measured by RT-qPCR. 2B11 and 2G8 inhibited the '\n",
      " 'viral entry into cells, indicating that the effect of these two nAbs was '\n",
      " 'mediated by the block of viral attachment to cells (Fig. 5a) . GDS01 and nAb '\n",
      " 'were added to Vero cells at separate time points to further confirm the '\n",
      " 'result. Firstly, PEDV was incubated with Vero cells at 4°C for 1 h, allowing '\n",
      " 'the attachment of the virus to the cell surface without entering. The cells '\n",
      " 'were rinsed by PBS to remove the unattached virus. Subsequently, nAbs were '\n",
      " 'added to the PEDV-cell complexes and the temperature was increased to 37°C '\n",
      " 'to initiate the viral fusion for 1 h. If the nAbs block viral attachment to '\n",
      " 'cells, pre-binding of PEDV to the cells before incubation with nAb would ) '\n",
      " 'produced by Baculovirus expression is shown on the below. SP is the S1 and '\n",
      " 'partly S2 gene fragment (encoding 1-954aa). The experiment was repeated two '\n",
      " 'times, and representative images are shown Fig. 4 Virus neutraliztion test '\n",
      " 'against PEDV-GDS01 and PEDV-GDS03. a Neutralization titer of antibody '\n",
      " 'against GDS01. The titer of 2G8, 2B11, 1E3 and 3D9 was assessed by the '\n",
      " 'plaque reduction assay. The negative serum from normal mouse was set as a '\n",
      " 'negative control. b PEDV-GDS03infected cells in the viral inhibition assay '\n",
      " 'were detected by IFA. 9G11 targeted N protein was set a second antibody. '\n",
      " 'Positive group used a negative serum and Negative group used a positive '\n",
      " 'serum as control respectively. All graphs represent the means from three '\n",
      " 'independent experiments. Error bars indicate standard deviations prevent '\n",
      " 'neutralization. Addition of the two nAbs after PEDV attachment to cells did '\n",
      " 'not reduce cell-associated PEDV (Fig. 5a) , suggesting that 2B11 and 2G8 '\n",
      " 'inhibit the infection of PEDV by blocking virus attachment to cells and not '\n",
      " 'downstream processes. Both 2G8 and 2B11 exerted their neutralization effects '\n",
      " 'by directly inhibiting the tested viruses binding to the infected cells. We '\n",
      " 'then determine whether the 2G8 and 2B11 antibodies bind to the same epitope. '\n",
      " 'As Fig. 5b shown in competition ELISA assay, 2B11 did not compete with 2G8 '\n",
      " 'for binding to PEDV. These data indicated the antibodies bind to distinct '\n",
      " 'epitopes. Multiple alphacoronaviruses, such as the TGEV, PRCoV, feline '\n",
      " 'coronavirus type II and human coronavirus 229E(HCoV-229E), used '\n",
      " 'aminopeptidase N (APN) as a receptor. But APN is not a universal receptor '\n",
      " 'for the alphacoronaviruses as the human coronavirus NL63(HCoV-NL63) used '\n",
      " 'angiotensin converting enzyme2(ACE2) for its entry [31, 32] . Presently, it '\n",
      " 'is believed [33] [34] [35] that porcine APN acts as a functional PEDV '\n",
      " 'receptor, however, whether or not pAPN is a receptor for PEDV has been '\n",
      " 'debated over the years [36, 37] . Intriguingly, Vero cell lines used for '\n",
      " \"isolation of PEDV strains don't express APN that inferred from the Vero cell \"\n",
      " 'proteome [38] . Some data indicated that other receptors may be involved in '\n",
      " 'PEDV entry into these cells, such as sialic acid and Neu5Ac [9, 39] . '\n",
      " 'Isolation a nAb that inhibit virus attachment to the cell surface could help '\n",
      " 'to identify the PEDV receptor. In this study, we screened 10 mAbs through '\n",
      " 'hybridoma technology. The main structural proteins of PEDV were expressed '\n",
      " 'using prokaryotic and eukaryotic expression system respectively. Because '\n",
      " 'immunogenic proteins were whole virus particles, the determination of the '\n",
      " 'target protein of mAb is a challenge. Prokaryotic expression system '\n",
      " 'expresses the products without any modification and its products are linear '\n",
      " 'proteins. Baculovirus expression system has the ability to express products '\n",
      " 'with glycosylation, phosphorylation and other processing modification after '\n",
      " 'translation, which are similar to natural proteins. 2B11, 1E3, 2B5, 2G10 and '\n",
      " '1A5 recognized the expressed SP protein specifically in sf9 cells but did '\n",
      " 'not bind to the SP protein expressed by BL21 cells. 1D11 recognized the '\n",
      " \"expressed N protein in sf9 cells but didn't bind to the N protein expressed \"\n",
      " 'by BL21 cells. The results indicated that 2B11, 1E3, 2B5, 2G10, 1A5 and 1D11 '\n",
      " 'specifically recognized the conformational epitope instead of the linearized '\n",
      " 'epitope. 9G11 and 3F10 recognized the linearized epitope, and 2G8 and 3D9 '\n",
      " \"had no reactivity with any expressed proteins. It's possible that 2G8 and \"\n",
      " '3D9 only recognize the trimer of S protein or S2 protein. Coronavirus '\n",
      " 'neutralization by antibodies is often attributed to antibody occupancy of '\n",
      " 'the S trimers and interfering with viral attachment to target cells or '\n",
      " 'entry. In addition, their neutralizing activity was exhibited in a '\n",
      " 'dose-dependent manner. 2G8 and 2B11 have high efficiency neutralization '\n",
      " '(IC50 < 10 μg/mL), 3D9 and 1E3 have moderate neutralization (10 μg/mL < IC50 '\n",
      " '< 100 μg/ mL). The observations clearly define the SP domain is most '\n",
      " 'critical for PEDV to interact with its target cells. Then anti-SP mAb 2B11 '\n",
      " 'and 2G8 with the strongest neutralizing capacity were selected to explore '\n",
      " 'the mechanism of nAbs. As previously reported, PEDV For the 37°C attachment, '\n",
      " 'PEDV was first incubated with antibody for 1 h and then added to cells for 1 '\n",
      " 'h infection incubation at 37°C. For the 4°C attachment, PEDV was initially '\n",
      " 'incubated with Vero cells for 1 h at 4°C, and antibody was then added to the '\n",
      " 'PEDV-cell culture and the temperature was increased to 37°C for 1 h '\n",
      " 'infection incubation. The cell-associated PEDV was quantified by viral RNA '\n",
      " 'RT-qPCR. 9G11 was set as an isotype control. b Comparison of epitopes '\n",
      " 'between 2G8 and 2B11 nAbs using a competition ELISA test. 2G8 was used as a '\n",
      " 'competitor, and 9G11 was used as a negative control. The results are '\n",
      " 'presented as the mean ± SEM (n = 3). Differences were considered significant '\n",
      " 'if the P value was < 0.05. P values are indicated as follows: *P < 0.05; '\n",
      " '***P < 0.001 enters Vero cells via an initial endocytic uptake, and '\n",
      " 'subsequently, the virus fuses with the PEDV S and host endosomal membrane '\n",
      " \"[40] . The virus only attaches cell, but doesn't have fusion with cell \"\n",
      " 'membrane at 4°C. We found 2G8 and anti-SP mAb 2B11 efficiently bound PEDV, '\n",
      " 'and then inhibited virus entry into cell at 37°C. But if the experiment was '\n",
      " 'designed into two-time points, virus infected the cells at 4°C for 1 h, and '\n",
      " 'then the mAb was added at 37°C for 1 h, the results showed that virus could '\n",
      " 'invade and replicate in cells, and the copies of virus in infected-cells had '\n",
      " 'no difference regardless of the presence of 2G8, anti-SP mAb 2B11 and '\n",
      " \"PEDV-negative serum. However, positive serum didn't prevent the \"\n",
      " 'proliferation of intracellular viruses. This may be due to the lack of mAb '\n",
      " 'in positive serum which neutralized the virus inside the cell or the '\n",
      " 'interaction of mAbs makes some mAbs lose the ability of neutralization '\n",
      " 'intracellular or there may exist other possible mechanisms. These results '\n",
      " 'demonstrated if the viral have attached to the target cells, neutralization '\n",
      " \"of 2G8 and anti-SP mAb 2B11 doesn't work. It seems that PEDV infected cells \"\n",
      " 'apparently lower at 4°C than 37°C regardless of any antibodies, indicating '\n",
      " 'that PEDV is more efficiently taken up by cells through endocytosis at 37°C '\n",
      " 'than at 4°C. This is consistent with that the virus uptake more efficiently '\n",
      " 'through endocytosis at 37°C than at 4°C, which was observed in Herpes '\n",
      " 'simplex virus 1 infection [41] . The epitope targeted by 2G8 is completely '\n",
      " 'distinct from anti-SP mAb 2B11, there may be at least two mechanisms '\n",
      " 'involved neutralization effects by directly inhibiting binding to an '\n",
      " 'epitope.')\n"
     ]
    }
   ],
   "source": [
    "pprint(co.text(comm_use[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'abstract', 'body_text', 'bib_entries', 'ref_entries', 'metadata', 'back_matter', 'paper_id'}\n"
     ]
    }
   ],
   "source": [
    "keys = comm_use.apply(lambda x: list(x.keys()))\n",
    "# then lets combine them into a set\n",
    "print(set(sum(keys, [])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
